a10_octavia/controller/worker/tasks/vthunder_tasks.py

```diff
@@ -359,3 +360,212 @@ def revert(self, vthunder, *args, **kwargs):
         except Exception as e:
             LOG.exception("Failed to revert partition create : %s", str(e))
             raise
+
+
+class TagEthernetBaseTask(task.Task):
+
+    def __init__(self, **kwargs):
+        super(TagEthernetBaseTask, self).__init__(**kwargs)
+        self._network_driver = None
+        self._subnet = None
+        self._subnet_ip = None
+        self._subnet_mask = None
+
+    @property
+    def network_driver(self):
+        if self._network_driver is None:
+            self._network_driver = a10_task_utils.get_a10_network_driver()
+        return self._network_driver
+
+    def get_tag_interface_info(self, project_id, vlan_id):
+        if project_id in CONF.hardware_thunder.devices:
+            vthunder_conf = CONF.hardware_thunder.devices[project_id]
+            if vthunder_conf.interface_vlan_map:
+                interface_vlan_map = vthunder_conf.interface_vlan_map
+                for ifnum in interface_vlan_map:
+                    vlan_dict = interface_vlan_map[ifnum]
+                    vlan_id_str = str(vlan_id)
+                    if vlan_id_str in vlan_dict:
+                        return (int(ifnum), vlan_dict[vlan_id_str])
+        return (1, None)
+
+    def get_subnet_and_mask(self, subnet_id):
```

Comment:
If we are getting things, then we should be returning them instead relying on a side effect. Could be dangerous if used in multiple places.